src/Common/HostResolvePool.h

```diff
@@ -141,6 +142,7 @@ class HostResolver : public std::enable_shared_from_this<HostResolver>
         size_t usage = 0;
         bool failed = false;
         Poco::Timestamp fail_time = 0;
+        size_t fail_count = 0;
```

Comment:
if you what to count failtures that you do not need this at all `bool failed = false;`

Reply:
uint_8 fail_count = 0;
 + 
static_assert that (1 << sizeof (fail_count)) <= RECORD_FAIL_COUNT_LIMIT, or something like that.

Reply:
It;s a different. `failed` flag cleaned from time to time (HostResolverPool.cpp, line 248), and without `fail_count` it is cleaned every 2 minutes (`DEFAULT_RESOLVE_TIME_HISTORY_SECONDS`). It is possible that in some time IP can be accessible again. `failed` set to `true` to check address again, but `fail_count` is not set to zero to make pauses between checks in 2, 4, 8, ..., 62 minutes. `fail_count` set to zero only after success check.

Reply:
I understood that too after some time ) Tnx.
May be it is better name for `fail_count` as `consecutive_fail_count`. You make bigger penalty according consecutive failures.